AK/BitStream.h

```diff
@@ -14,6 +14,41 @@
 
 namespace AK {
 
+class BitStream : public Stream {
+protected:
+    using BufferType = u64;
+
+    static constexpr size_t bits_per_byte = 8u;
+    static constexpr size_t bit_buffer_size = sizeof(BufferType) * bits_per_byte;
+
+    explicit BitStream(MaybeOwned<Stream> stream)
+        : m_stream(move(stream))
+    {
+    }
+
+    template<Unsigned T>
+    static constexpr T lsb_mask(T bits)
```

Comment:
Why is it called lsb? I guess that the class name has no mention of `LittleEndian` because it could be shared with `BigEndian`, so we can maybe rename that to `bit_mask`?

Reply:
Fair point, it's called lsb just because it's moved from the little endian class that called it lsb. I'll rename this base class to be LittleEndianBitStream for now because I'm not actually sure how much of it will apply to big endian, there's other "lsb" methods here too.